FlowCal/mef.py

```diff
@@ -265,27 +320,34 @@ def fit_beads_autofluorescence(fl_channel, fl_mef):
     # Initial guesses:
     # 0: slope found by putting a line through the highest two points.
     # 1: y-intercept found by putting a line through highest two points.
-    # 2: bead autofluorescence initialized to 100.
+    # 2: bead autofluorescence initialized using the first point.
     params[0] = (np.log(fl_mef[-1]) - np.log(fl_mef[-2])) / \
-                    (fl_channel[-1] - fl_channel[-2])
-    params[1] = np.log(fl_mef[-1]) - params[0] * fl_channel[-1]
-    params[2] = 100.
+                (np.log(fl_au[-1]) - np.log(fl_au[-2]))
+    params[1] = np.log(fl_mef[-1]) - params[0] * np.log(fl_au[-1])
+    params[2] = np.exp(params[0]*np.log(fl_au[0]) + params[1]) - fl_mef[0]
```

Comment:
The new equation is a better estimate for `params[2]`.

(In reality, I changed this because the fitting function was not working at some point, and I thought this was the problem. Turns out I had a typo somewhere else, but then I decided to leave the new estimate).

If fluorescence of the populations looks as we expect, I think using the last two peaks is better than   fitting a line. If fluorescence is different than we expect, you might be right that fitting a line is more robust. However, all the test data that I have show that fitting is working well. Changing this should be a new issue, and I could give you all the test data if you want to show that this change will improve things. I'm not particularly interested as I don't think this is a problem, though.